torch/csrc/api/include/torch/optim/schedulers/reduce_on_plateau_scheduler.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <torch/optim/optimizer.h>
+
+#include <torch/csrc/Export.h>
+
+#include <string>
+
+#include <cmath>
+
+#include <iostream>
+
+namespace torch {
+namespace optim {
+
+class TORCH_API ReduceLROnPlateauScheduler {
```

Comment:
this should derive from the base `LRScheduler`, no?

Reply:
Thanks for the comment! Actually i am not really sure if this is a good idea, it will violate a bit liskov substitution principle - since interface is a bit different for both schedulers - step(void) vs step(float). As far as i am concerned, this is also the arrangement in Python API.